specification/under_development/profiles/Adaptive_Profile.md

```diff
@@ -0,0 +1,1028 @@
+!9 = !{i32 1, !"backwards_branching", i1 false}
+!10 = !{i32 1, !"multiple_target_branching", i1 false}
+```
+
+The program performs gate teleportation, and it uses conditional single qubit
+gates and mid-circuit measurements to effect control flow.
+
+For the sake of clarity, the code above does not contain any [debug
+symbols](https://releases.llvm.org/13.0.0/docs/tutorial/MyFirstLanguageFrontend/LangImpl09.html?highlight=debug%20symbols).
+Debug symbols contain information that is used by a debugger to relate failures
+during execution back to the original source code. While we expect this to be
+primarily useful for execution on simulators, debug symbols are both easy to
+ignore and may be useful to generate helpful error messages for compilation
+failures that happen only late in the process. We hence see no reason to
+disallow them from occurring in the bitcode but will not detail their use any
+further. We defer to existing resources for more information about how to
+generate and use debug symbols.
+
+## Entry Point Definition
+
+The bitcode contains the definition of the LLVM function that should be invoked
+when the program is executed, referred to as "entry point" in the rest of this
+profile specification. The name of this function may be chosen freely, as long
+as it is a valid [global
+identifier](https://llvm.org/docs/LangRef.html#identifiers) according to the
+LLVM standard. The entry point is identified by a custom function attribute; the
+section on [attributes](#attributes) defines which attributes must be attached
+to the entry point function.
+
+An entry point function may not take any parameters and must either return void
+in case the program has no failure conditions or must return an exit code in the
+form of a 64-bit integer. In the case that a shot fails either because the entry
+point function returns an integer value indicating that the program caught an
+error or because an instruction caused an uncaught error (for example a `div`
+instruction fails due to division by 0), then the output format should indicate
+that the shot failed with output for the shot appearing as `ERROR Code: ival`
+where `ival` is the error code from the adaptive profile program or `ERROR message`
+in the case that an uncaught failure occurs, and the message is chosen by the backend.
+
+The adaptive profile program makes no restrictions on the structure of basic
+blocks within the entry point function, other than that a block cannot jump to a
+previously encountered block in the control flow graph unless a backend opts
+into **Bullet 11**. By default adaptive profile programs limit branching to only
+express forward branching and nested conditionality. Additionally, the only
+functions that can be called by default in the entry block are `qis` or `rt`
+functions defined in the instruction set and profile. This restriction is removed
+if a backend opts into **Bullet 10** which allows for user-defined functions.
```

Comment:
[This example](https://github.com/qir-alliance/qir-spec/blob/93597a337816e260628eb4d53b8510427251d4a9/specification/under_development/profiles/Adaptive_Profile.md?plain=1#L324-L327) from Bullet 9 makes me assume that classical instructions are also allowed. If so, I would suggest to change the last sentence of this paragraph to something like this:

> This restriction is removed if a backend opts into Bullet 9 or 10 which allow for classical instructions and user-defined functions, respectively.